### PR TITLE
Add a number of fixers

### PR DESCRIPTION
**Pull request by dakota**
This adds some fixers for some of the internal sniffs..

### CODE REVIEW

File: CakePHP/Sniffs/NamingConventions/ValidVariableNameSniff.php [hunk 1]
```diff
@@ -147,21 +150,45 @@ protected function processVariableInString(PHP_CodeSniffer_File $phpcsFile, $sta
...
+        $phpcsFile->fixer->endChangeset();
+    }
+
+    /**
+     * Check that a variable is a valid shape.
+     *
+     * Variables in CakePHP can either be $fooBar, $FooBar, $_fooBar, or $_FooBar.
+     *
+     * @param string $string The variable to check.
+     * @param boolea $public Whether or not the variable is public.
```
Comment (dereuromark) [inline CakePHP/Sniffs/NamingConventions/ValidVariableNameSniff.php]: Bool

**Comment by lorenzo** [comment]
Good job!

**Comment by dereuromark** [comment]
The sniffer should also Auto phpcs check itself imo.

**Comment by dereuromark** [comment]
PS: you might also want to port some of my powerful new [PSR2R sniffer auto-fixers](https://github.com/php-fig-rectified/psr2r-sniffer).
Specifically ordered use and unused use as well as some other useful ones.

**Pull request merged**